src/features/speedup/components/SpeedUpModal.tsx

```diff
@@ -0,0 +1,148 @@
+import useGasPrice from '@/hooks/useGasPrice'
+import useGasLimit from '@/hooks/useGasLimit'
+import ModalDialog from '@/components/common/ModalDialog'
+import DialogContent from '@mui/material/DialogContent'
+import { Box, Button, SvgIcon, Tooltip, Typography } from '@mui/material'
+import RocketSpeedup from '@/public/images/common/ic-rocket-speedup.svg'
+import DialogActions from '@mui/material/DialogActions'
+import { useSafeTransaction } from '@/features/speedup/hooks/useSafeTransaction'
+import useWallet from '@/hooks/wallets/useWallet'
+import useOnboard from '@/hooks/wallets/useOnboard'
+import useChainId from '@/hooks/useChainId'
+import useSafeAddress from '@/hooks/useSafeAddress'
+import { useAppDispatch } from '@/store'
+import { dispatchTxSpeedUp } from '@/services/tx/tx-sender'
+import { showNotification } from '@/store/notificationsSlice'
+import { useCallback, useMemo, useState } from 'react'
+import GasParams from '@/components/tx/GasParams'
+import { type TransactionDetails } from '@safe-global/safe-gateway-typescript-sdk'
+import { asError } from '@/services/exceptions/utils'
+
+type Props = {
+  open: boolean
+  handleClose: () => void
+  txDetails: undefined | TransactionDetails
+  txId: string
+  signerAddress: string | undefined
+  signerNonce: number | undefined | null
+}
+export const SpeedUpModal = ({ open, handleClose, txDetails, txId, signerAddress, signerNonce }: Props) => {
+  const [gasPrice] = useGasPrice()
+  const [waitingForConfirmation, setWaitingForConfirmation] = useState(false)
+  const safeTx = useSafeTransaction(txDetails)
+  const { gasLimit } = useGasLimit(safeTx)
+
+  const wallet = useWallet()
+  const onboard = useOnboard()
+  const chainId = useChainId()
+  const safeAddress = useSafeAddress()
+  const hasActions = signerAddress && signerNonce !== undefined && signerAddress === wallet?.address
+  const dispatch = useAppDispatch()
+
+  const isDisabled = waitingForConfirmation || !wallet || !safeTx || !gasPrice || !txDetails || !onboard
+
+  const speedUpFee = useMemo(
+    () => ({
+      ...gasPrice,
+      maxFeePerGas: BigInt(gasPrice?.maxFeePerGas ?? 10n) + BigInt(gasPrice?.maxPriorityFeePerGas ?? 1n),
```

Comment:
We have a feature toggle for EIP 1559. 
If this is turned on we provide `maxFeePerGas` and `maxPriorityFeePerGas`
otherwise just a `gasPrice`.

(See the useGasPrice hook)

It might be easier to add a new boolean to the useGasPrice hook `speedUp: boolean` or `speedUpFactor: number` and multiply the fee there.